include/tmp/export

```diff
@@ -0,0 +1,21 @@
+#ifndef TMP_EXPORT_H
+#define TMP_EXPORT_H
+
+#if defined _WIN32
+#if defined TMP_BUILDING_DLL
+#define TMP_EXPORT __declspec(dllexport)
+#elif defined TMP_SHARED
+#define TMP_EXPORT __declspec(dllimport)
+#else
+#define TMP_EXPORT
+#endif
+#define TMP_NO_EXPORT
+#elif defined TMP_BUILDING_DLL || defined TMP_SHARED
+#define TMP_EXPORT __attribute__((visibility("default")))
+#define TMP_NO_EXPORT __attribute__((visibility("hidden")))
+#else
+#define TMP_EXPORT
+#define TMP_NO_EXPORT
+#endif
```

Comment:
Note that on Linux it is never problematic to define the visibility attributes even for static builds.